apps/client/src/components/UI/AccessibleNavigation/AccessibleNavigation.tsx

```diff
@@ -0,0 +1,171 @@
+import { Slot } from "@radix-ui/react-slot";
+import React, {
+  KeyboardEvent,
+  ReactNode,
+  createContext,
+  forwardRef,
+  memo,
+  useCallback,
+  useContext,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from "react";
+
+type Orientation = "horizontal" | "vertical";
+
+type AccessibleNavigationProps = React.HTMLAttributes<HTMLDivElement> & {
+  "children": ReactNode;
+  "orientation"?: Orientation;
+  "onEscape"?: () => void;
+  "onNavigateOut"?: () => void;
+  "aria-label"?: string; // Accessible name for the navigation
+};
+
+type AccessibleNavigationContextProps = {
+  activeIndex: number;
+  focusItem: (index: number) => void;
+};
+
+const AccessibleNavigationContext = createContext<AccessibleNavigationContextProps | undefined>(undefined);
+
+const AccessibleNavigation = forwardRef<HTMLDivElement, AccessibleNavigationProps>(
+  (
+    { children, orientation = "vertical", className, onEscape, onNavigateOut, "aria-label": ariaLabel, ...props },
+    ref,
+  ) => {
+    const navRef = useRef<HTMLDivElement | null>(null);
+    const [activeIndex, setActiveIndex] = useState(0);
+
+    useImperativeHandle(ref, () => navRef.current as HTMLDivElement);
```

Comment:
This usage of useImperativeHandle doesn't seem to acheive much.  Unless I'm missing something might as well use the ref directly.

Reply:
Thanks for your eagle-eye ^^
I just made a few tests to be sure.
If I don't use the imperativeHandle react loose the underlying div and the roving index mechanism won't work as expected